pkg/controllers/node/termination/terminator/eviction.go

```diff
@@ -192,32 +176,42 @@ func (q *Queue) Evict(ctx context.Context, key QueueKey) bool {
 		}
 		// status codes for the eviction API are defined here:
 		// https://kubernetes.io/docs/concepts/scheduling-eviction/api-eviction/#how-api-initiated-eviction-works
-		if apierrors.IsNotFound(err) || apierrors.IsConflict(err) {
-			// 404 - The pod no longer exists
-			// https://github.com/kubernetes/kubernetes/blob/ad19beaa83363de89a7772f4d5af393b85ce5e61/pkg/registry/core/pod/storage/eviction.go#L160
-			// 409 - The pod exists, but it is not the same pod that we initiated the eviction on
-			// https://github.com/kubernetes/kubernetes/blob/ad19beaa83363de89a7772f4d5af393b85ce5e61/pkg/registry/core/pod/storage/eviction.go#L318
-			return true
+		// 404 - The pod no longer exists
+		// https://github.com/kubernetes/kubernetes/blob/ad19beaa83363de89a7772f4d5af393b85ce5e61/pkg/registry/core/pod/storage/eviction.go#L160
+		// 409 - The pod exists, but it is not the same pod that we initiated the eviction on
+		// https://github.com/kubernetes/kubernetes/blob/ad19beaa83363de89a7772f4d5af393b85ce5e61/pkg/registry/core/pod/storage/eviction.go#L318
+		if !(apierrors.IsNotFound(err) || apierrors.IsConflict(err)) {
```

Comment:
I think we should just exit immediately if we have a pod that no longer exists or is a different pod than the one that we intended to evict

Reply:
As it stands right now, you are adding metrics and logging that you are evicting pods when you are not actually doing any kind of eviction